code/modules/power/cable.dm

```diff
@@ -130,7 +133,7 @@ By design, d1 is the smallest direction and d2 is the highest
 /obj/structure/cable/attackby(obj/item/W, mob/user)
 	var/turf/T = src.loc
 
-	if(T.intact)
+	if(isturf(T) && T.intact) //Exception for catwalks

```

Comment:
Shouldn't be required.
Adding cable to a catwalk wouldn't actually make the cable's loc the catwalk, unless I misunderstand your code and you're placing cable inside the catwalk.